src/mscorlib/corefx/System/Globalization/TextInfo.Unix.cs

```diff
@@ -36,9 +36,35 @@ private unsafe string ChangeCase(string s, bool toUpper)
             }
 
             string result = string.FastAllocateString(s.Length);
-            fixed (char* pBuf = result)
+
+            fixed (char* pResult = result)
             {
-                Interop.GlobalizationInterop.ChangeCase(s, s.Length, pBuf, result.Length, toUpper, m_needsTurkishCasing);
+                if (IsAsciiCasingSameAsInvariant && s.IsAscii())
+                {
+                    fixed (char* pSource = s)
```

Comment:
Thanks.  I'd prefer to leave it as-is and refactor them out if/when we find that necessary.